app/routes/user/index.tsx

```diff
@@ -7,21 +7,68 @@
  */
 
 import type { DataFunctionArgs } from '@remix-run/node'
-import { useLoaderData } from '@remix-run/react'
+import { redirect } from '@remix-run/node'
+import { Form, useLoaderData } from '@remix-run/react'
+import { Button, Label, TextInput } from '@trussworks/react-uswds'
+import { getFormDataString } from '~/lib/utils'
 import { getUser } from '../__auth/user.server'
+import { UserDataServer } from './user_data.server'
 
 export async function loader({ request }: DataFunctionArgs) {
   const user = await getUser(request)
   if (!user) throw new Response(null, { status: 403 })
-  return { email: user.email, idp: user.idp }
+  const server = await UserDataServer.create(request)
+  const addlUserData = await server.getUserData()
+  return {
+    email: user.email,
+    idp: user.idp,
+    username: addlUserData.username,
+    association: addlUserData.association,
+  }
+}
+
+export async function action({ request }: DataFunctionArgs) {
+  const data = await request.formData()
+  const username = getFormDataString(data, 'username')
+  const association = getFormDataString(data, 'association')
+
+  if (!username) throw new Response('username not set', { status: 400 })
+  if (!association) throw new Response('association not set', { status: 400 })
+
+  const server = await UserDataServer.create(request)
+  await server.updateUserData(username, association)
+
+  return redirect('/user')
 }
 
 export default function User() {
-  const { email, idp } = useLoaderData<typeof loader>()
+  const { email, idp, username, association } = useLoaderData<typeof loader>()
   return (
     <>
       <h1>Welcome, {email}!</h1>
       <p>You signed in with {idp || 'username and password'}.</p>
+      <h2>Profile Info</h2>
+      <Form method="post">
+        <Label htmlFor="username">Display Name</Label>
```

Comment:
```suggestion
        <Label htmlFor="username">Display Name</Label>
```

Add some explanatory text to the effect of: "How would you like your name to appear in GCN Circulars? For example: A. E. Einstein, A. Einstein, Albert Einstein"

Reply:
Good call, will have this updated in the next commit